airflow/models/taskinstance.py

```diff
@@ -1110,6 +1116,27 @@ def signal_handler(signum, frame):
             session.merge(self)
         session.commit()
 
+    def get_additional_execution_contextmanager(self, execution_context):
+        """
+        Retrieves the user defined execution context callback from the configuration,
+        and validates that it is indeed a context manager
+
+        :param execution_context: the current execution context to be passed to user ctx
+        """
+        additional_execution_contextmanager = conf.getimport("core", "additional_execute_contextmanager")
+        if additional_execution_contextmanager:
+            try:
+                user_ctx_obj = additional_execution_contextmanager(self, execution_context)
+                if hasattr(user_ctx_obj, "__enter__") and hasattr(user_ctx_obj, "__exit__"):
+                    return user_ctx_obj
+                else:
+                    raise AirflowException(f"Loaded function {additional_execution_contextmanager} "
+                                           f"as additional execution contextmanager, but it does not have "
+                                           f"__enter__ or __exit__ method!")
```

Comment:
What about `@contextlib.contextmanager`? I think this check works only for context managers defined as a class

Reply:
The tests verify that this works for decorated contextmanagers as well :)

Reply:
Good!